contracts/InterchainTokenService.sol

```diff
@@ -415,8 +414,20 @@ contract InterchainTokenService is
 
         token.safeTransferFrom(msg.sender, destinationAddress, amount);
 
-        if (messageType == MESSAGE_TYPE_INTERCHAIN_TRANSFER_WITH_DATA) {
-            (, , , , , bytes memory data) = abi.decode(payload, (uint256, bytes32, bytes, bytes, uint256, bytes));
+        if (data.length == 0) {
+            // slither-disable-next-line reentrancy-events
+            emit InterchainTransferReceived(commandId, tokenId, sourceChain, sourceAddress, destinationAddress, amount);
+        } else {
+            // slither-disable-next-line reentrancy-events
+            emit InterchainTransferReceivedWithData(
```

Comment:
combine the events for simplicity? Should be easy to test for `keccak256(0)` (if data is empty, could use a precomputed `keccak256(0)` constant)?

Reply:
Let's keep events separate as we have 2 separate events when transfer arrives to the destination and represent different behavior